cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/InMemoryConfigurator.java

```diff
@@ -113,14 +117,16 @@ public InMemoryConfigurator(Location archive) {
   }
 
   private ConfigResponse createResponse(Application app, String bundleVersion) {
-    String specJson = getSpecJson(app, bundleVersion);
+    String specJson = getSpecJson(app, bundleVersion, configString);
     return new DefaultConfigResponse(0, CharStreams.newReaderSupplier(specJson));
   }
 
-  private static String getSpecJson(Application app, final String bundleVersion) {
+  private static String getSpecJson(Application app, final String bundleVersion, final String configString) {
     // Now, we call configure, which returns application specification.
     DefaultAppConfigurer configurer = new DefaultAppConfigurer(app);
-    app.configure(configurer, new DefaultApplicationContext());
+    TypeToken typeToken = TypeToken.of(app.getClass());
+    TypeToken<?> resultToken = typeToken.resolveType(Application.class.getTypeParameters()[0]);
```

Comment:
it will work as long as user doesn't provide any configuration string in the header. if he does, then an GSON deserialization will fail since it will try to create a Config object with the config string provided.